simple_certmanager/admin_views.py

```diff
@@ -0,0 +1,145 @@
+            "organization_name",
+            "state_or_province_name",
+            "email_address",
+            "common_name",
+        ]
+        widgets = {
+            "country_name": forms.TextInput(attrs={"class": "form-control"}),
+            "organization_name": forms.TextInput(attrs={"class": "form-control"}),
+            "state_or_province_name": forms.TextInput(attrs={"class": "form-control"}),
+            "email_address": forms.EmailInput(attrs={"class": "form-control"}),
+            "common_name": forms.TextInput(attrs={"class": "form-control"}),
+        }
+
+
+class GenerateCertificateView(generic.edit.CreateView):
+    template_name_suffix = "_form"
+    model = CertificateSigningRequestInfo
+    form_class = CertificateSigningRequestInfoForm
+    success_url = "/admin/simple_certmanager/certificate/"
+
+    def form_valid(self, form):
+        results = {
+            "country": form.cleaned_data["country_name"],
+            "organization": form.cleaned_data["organization_name"],
+            "state": form.cleaned_data["state_or_province_name"],
+            "email_address": form.cleaned_data["email_address"],
+            "common_name": form.cleaned_data["common_name"],
+        }
+
+        if not results["common_name"].startswith("saml."):
+            results["common_name"] = "saml." + results["common_name"]
+
+        try:
+            private_key, csr = self.create_csr(
+                common_name=results["common_name"],
+                country=results["country"],
+                state=results["state"],
+                organization=results["organization"],
+                email_address=results["email_address"],
+            )
+
+            self.save_certificate(results, private_key, csr)
+        except Exception as e:
+            form.add_error(None, str(e))
+            return self.form_invalid(form)
+
+        return super().form_valid(form)
+
+    def save_certificate(self, results, private_key, csr):
+        private_key_file = ContentFile(private_key, name="private_key.pem")
+        csr_file = ContentFile(csr, name="csr.pem")
+
+        # Save the private key and CSR to the right location
+        info = CertificateSigningRequestInfo.objects.create(
+            country_name=results["country"],
+            organization_name=results["organization"],
+            state_or_province_name=results["state"],
+            email_address=results["email_address"],
+            common_name=results["common_name"],
+        )
+        db_certificate = Certificate.objects.create(
+            type=CertificateTypes.cert_only,
+            private_key=private_key_file,
+            csr=csr_file,
+            info=info,
+        )
+        db_certificate.label = (
+            f"#{db_certificate.id} - {results['organization']} certificate"
+        )
+        db_certificate.save()
+
+    def get_context_data(self, **kwargs):
+        return super().get_context_data(**kwargs)
+
+    def create_csr(
```

Comment:
IMO this belongs in `simple_certmanager/utils.py` so that it can easily be unit tested and invoked from the `models.py`.

I think I'd even go further and create a dedicated module for it: `simple_certmanager/csr.py`